src/sentry/search/snuba/executors.py

```diff
@@ -472,6 +457,323 @@ def query(
+
+    dependency_aggregations = {"priority": ["events.last_seen", "times_seen"]}
+    issue_only_fields = set(
+        ["query", "bookmarked_by", "assigned_to", "unassigned", "subscribed_by"]
+    )
+    sort_strategies = {
+        # TODO: If not using environment filters, could these sort methods use last_seen and first_seen from groups instead? so only add prefix conditionally?
+        "date": "events.last_seen",
+        "freq": "times_seen",
+        "new": "events.first_seen",
+        "priority": "priority",
+    }
+
+    aggregation_defs = {
+        "times_seen": ["count()", ""],
+        "events.first_seen": ["multiply(toUInt64(min(events.timestamp)), 1000)", ""],
+        "events.last_seen": ["multiply(toUInt64(max(events.timestamp)), 1000)", ""],
+        "priority": ["toUInt64(plus(multiply(log(times_seen), 600), `events.last_seen`))", ""],
+        "total": ["uniq", ISSUE_FIELD_NAME],
+    }
+
+    def _transform_converted_filter(
+        self, search_filter, converted_filter, project_ids=None, environment_ids=None
+    ):
+        converted_name = search_filter.key.name
+
+        special_date_names = ["active_at", "first_seen", "last_seen"]
+        if search_filter.key.name in special_date_names:
+            # Need to get '2018-02-06T03:35:54' format out of 1517888878000 format
+            datetime_value = datetime.datetime.fromtimestamp(converted_filter[2] / 1000)
+            datetime_value = datetime_value.replace(microsecond=0).isoformat().replace("+00:00", "")
+            converted_filter[2] = datetime_value
+
+        # TODO: There is a better way to do this...the issue is that the table/alias is forked on environments, which can't be used in constrain_column_to_dataset
+        if search_filter.key.name in ["first_seen", "last_seen"]:  # , "first_release"]:
+            if environment_ids is not None:
+                table_alias = "events."
+                # return None, None #??? remove as a filter, since we are going to pre/post filter?
+            else:
+                table_alias = "groups."
+
+            if isinstance(converted_filter[0], list):
+                converted_filter[0][1][0] = table_alias + converted_filter[0][1][0]
+                converted_name = converted_filter[0][1][0]
+            else:
+                converted_filter[0] = table_alias + converted_filter[0]
+                converted_name = converted_filter[0]
+
+        if search_filter.key.name == "first_release":
+            # The filter's value will be the release's "version". Snuba only knows about ID. So we convert version to id here.
+            release = Release.objects.filter(
+                version=converted_filter[2],
+                organization_id=Project.objects.get(id=project_ids[0]).organization_id,
+            )
+            if not release:
+                # TODO: This means there will be no results and we do not need to run this query!
```

Comment:
It seems like get_snuba_translators is designed to work with .filter_keys, and not .conditions. It currently doesn't seem to process conditions at all, but maybe I missed something?

Reply:
Good point... One option is to pass it as a filter_key instead and just reuse this logic

Reply:
I now do this translation in the query function, so that if the mapping fails, we can return an empty result set before ever running the snuba query in the first place. What do you think of this?